### PR TITLE
node: move the reboot detection logic to kubelet

### PR DESCRIPTION
**Pull request by simon3z**

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1752,8 +1752,20 @@ func (kl *Kubelet) tryUpdateNodeStatus() error {
 	} else {
 		node.Status.NodeInfo.MachineID = info.MachineID
 		node.Status.NodeInfo.SystemUUID = info.SystemUUID
-		node.Status.NodeInfo.BootID = info.BootID
 		node.Status.Capacity = CapacityFromMachineInfo(info)
+		if node.Status.NodeInfo.BootID != "" &&
+			node.Status.NodeInfo.BootID != info.BootID {
+			ref := &api.ObjectReference{
+				Kind:      "Minion",
```
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: We use Node in the BirthCry (worrysome that we had both references)..

**Comment by simon3z** [comment]
@vmarmol now it should be ok.

**Comment by vmarmol** [comment]
Thanks @simon3z! This LGTM, just to verify since I am unfamiliar: boot ID changes after each boot?

**Comment by vmarmol** [comment]
Verified that was the case, merging.

**Pull request merged**